// @sanity/base/lib/components/ZIndexProvider.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useZIndex = useZIndex;
exports.ZIndexProvider = ZIndexProvider;

var _react = _interopRequireWildcard(require("react"));

var _sanityCssCustomProperties = _interopRequireDefault(require("sanity:css-custom-properties"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function getCustomCSSPropertyNumber(key) {
  var rawValue = _sanityCssCustomProperties.default[key];

  if (rawValue === undefined) {
    // eslint-disable-next-line no-console
    console.warn("getCustomCSSPropertyNumber: the custom CSS property \"".concat(key, "\" is not defined"));
    return undefined;
  }

  var value = parseInt(rawValue, 10);

  if (isNaN(value)) {
    // eslint-disable-next-line no-console
    console.warn("getCustomCSSPropertyNumber: the custom CSS property \"".concat(key, "\" is not a number"));
    return undefined;
  }

  return value;
}

var defaults = {
  navbar: 200,
  navbarPopover: 500000,
  navbarDialog: 500001,
  pane: 100,
  paneResizer: 150,
  portal: 200,
  popover: 200,
  modal: 200,
  movingItem: 10000,
  drawershade: 1000000,
  drawer: 1000001,
  // NOT IN USE
  dropdown: 200,
  navbarFixed: 1010,
  fullscreenEdit: 1050,
  popoverBackground: 1060,
  tooltip: 200,
  modalBackground: 2000,
  spinner: 3000
};

function getLegacyZIndexes() {
  return {
    /*
      used by
      - Navbar
    */
    navbar: getCustomCSSPropertyNumber('--zindex-navbar') || defaults.navbar,
    navbarPopover: getCustomCSSPropertyNumber('--zindex-navbar-popover') || defaults.navbarPopover,
    navbarDialog: getCustomCSSPropertyNumber('--zindex-navbar-dialog') || defaults.navbarDialog,

    /*
      used by:
      - DefaultPane
    */
    pane: getCustomCSSPropertyNumber('--zindex-pane') || defaults.pane,

    /*
      used by:
      - DefaultPane
    */
    paneResizer: getCustomCSSPropertyNumber('--zindex-pane-resizer') || defaults.paneResizer,

    /*
      used by:
      - EditItemFoldOut
      - Spinner
      - ConnectorsOverlay
      - tippy.css
      - BaseDateTimeInput
    */
    portal: getCustomCSSPropertyNumber('--zindex-portal') || defaults.portal,

    /*
      used by tooltip
    */
    popover: getCustomCSSPropertyNumber('--zindex-popover') || defaults.popover,

    /*
      used by google-maps-input
    */
    modal: getCustomCSSPropertyNumber('--zindex-modal') || defaults.modal,

    /*
      used for `movingItem` in:
      packages/@sanity/base/src/styles/layout/helpers.css
    */
    movingItem: getCustomCSSPropertyNumber('--zindex-moving-item') || defaults.movingItem,

    /*
      used for shadow behind the navbar search, and behind sidemenu
    */
    drawershade: getCustomCSSPropertyNumber('--zindex-drawershade') || defaults.drawershade,

    /*
      used for snackbar
    */
    drawer: getCustomCSSPropertyNumber('--zindex-drawer') || defaults.drawer,
    // NOT IN USE
    dropdown: getCustomCSSPropertyNumber('--zindex-dropdown') || defaults.dropdown,
    navbarFixed: getCustomCSSPropertyNumber('--zindex-navbar-fixed') || defaults.navbarFixed,
    fullscreenEdit: getCustomCSSPropertyNumber('--zindex-fullscreen-edit') || defaults.fullscreenEdit,
    popoverBackground: getCustomCSSPropertyNumber('--zindex-popover-background') || defaults.popoverBackground,
    tooltip: getCustomCSSPropertyNumber('--zindex-tooltip') || defaults.tooltip,
    modalBackground: getCustomCSSPropertyNumber('--zindex-modal-background') || defaults.modalBackground,
    spinner: getCustomCSSPropertyNumber('--zindex-spinner') || defaults.spinner
  };
}

var ZIndexContext = /*#__PURE__*/(0, _react.createContext)(defaults);

function useZIndex() {
  return (0, _react.useContext)(ZIndexContext);
}

function ZIndexProvider(_ref) {
  var children = _ref.children;
  var zIndexes = (0, _react.useMemo)(() => getLegacyZIndexes(), []);
  return /*#__PURE__*/_react.default.createElement(ZIndexContext.Provider, {
    value: zIndexes
  }, children);
}
