// @sanity/base/lib/components/formField/FormFieldHeader.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FormFieldHeader = FormFieldHeader;

var _react = _interopRequireDefault(require("react"));

var _ui = require("@sanity/ui");

var _presence = require("../../presence");

var _FormFieldHeaderText = require("./FormFieldHeaderText");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* eslint-disable camelcase */
function FormFieldHeader(props) {
  var markers = props.__unstable_markers,
      presence = props.__unstable_presence,
      description = props.description,
      inputId = props.inputId,
      title = props.title;
  return /*#__PURE__*/_react.default.createElement(_ui.Flex, {
    align: "flex-end"
  }, /*#__PURE__*/_react.default.createElement(_ui.Box, {
    flex: 1,
    paddingY: 2
  }, /*#__PURE__*/_react.default.createElement(_FormFieldHeaderText.FormFieldHeaderText, {
    __unstable_markers: markers,
    description: description,
    inputId: inputId,
    title: title
  })), presence && presence.length > 0 && /*#__PURE__*/_react.default.createElement(_ui.Box, null, /*#__PURE__*/_react.default.createElement(_presence.FieldPresence, {
    maxAvatars: 4,
    presence: presence
  })));
}
