// @sanity/base/lib/components/formField/FormFieldHeaderText.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FormFieldHeaderText = FormFieldHeaderText;

var _types = require("@sanity/types");

var _ui = require("@sanity/ui");

var _react = _interopRequireDefault(require("react"));

var _FormFieldValidationStatus = require("./FormFieldValidationStatus");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* eslint-disable camelcase */
function FormFieldHeaderText(props) {
  var description = props.description,
      inputId = props.inputId,
      title = props.title,
      _props$__unstable_mar = props.__unstable_markers,
      markers = _props$__unstable_mar === void 0 ? [] : _props$__unstable_mar;
  var validationMarkers = markers.filter(_types.isValidationMarker);
  var hasValidations = validationMarkers.length > 0;
  return /*#__PURE__*/_react.default.createElement(_ui.Stack, {
    space: 2
  }, /*#__PURE__*/_react.default.createElement(_ui.Flex, null, /*#__PURE__*/_react.default.createElement(_ui.Text, {
    as: "label",
    htmlFor: inputId,
    weight: "semibold",
    size: 1
  }, title || /*#__PURE__*/_react.default.createElement("em", null, "Untitled")), hasValidations && /*#__PURE__*/_react.default.createElement(_ui.Box, {
    marginLeft: 2
  }, /*#__PURE__*/_react.default.createElement(_FormFieldValidationStatus.FormFieldValidationStatus, {
    fontSize: 1,
    __unstable_markers: markers
  }))), description && /*#__PURE__*/_react.default.createElement(_ui.Text, {
    muted: true,
    size: 1
  }, description));
}
